test/quantization/test_quantize_fx.py

```diff
@@ -757,3 +768,177 @@ def forward(self, x):
+        model_list -= {'shufflenet_v2_x1_0', 'mobilenet_v2'}
+
+        # mobilenet: dropout error RuntimeError: "bernoulli_scalar_cpu_" not implemented for 'QUInt8'
+        # incpetion_v3: looks like there is some problem with AuxLogits
+        quantized_not_working = [('qat', 'mobilenet_v2'),
+                                 ('qat', 'inception_v3'),
+                                 ('static', 'inception_v3')]
+
+        fx_eager_not_matching = ['googlenet',  # because _transform_input is not quantized in eager
+                                 'mobilenetv2']  # because relu6 is replaced as relu in mobilenetv2
+
+        input_tensor = torch.rand(1, 3, 224, 224)
+        input_tensor_inception = torch.rand(1, 3, 299, 299)
+        output_value = torch.randint(0, 1, (1,))
+
+        qconfig_dict = {'': torch.quantization.default_qconfig}
+        diff_of_quant = {}
+        diff_from_eager = {}
+        greater_than_one = {}
+        # 'ddp' is not supported yet since GraphModule can't be pickled
+        modes = ['static', 'qat']
+
+        options = itertools.product(modes, model_list)
+        for mode, name in options:
+            if mode not in diff_of_quant:
+                diff_of_quant[mode] = {}
+                diff_from_eager[mode] = {}
+                greater_than_one[mode] = {}
+
+            # print('quantizing:', name, ' mode:', mode)
+            pretrained = name in quantized_model_list  # load pretrained model to compare with quantized model
+            model = models.__dict__[name](pretrained=pretrained).eval().float()
+            if name in set(['inception_v3']):
+                input_value = input_tensor_inception
+            else:
+                input_value = input_tensor
+
+            qconfig = default_qconfig if mode == 'static' else default_qat_qconfig
+            qconfig_dict = {'': qconfig}
+            graph_module = symbolic_trace(model)
+            # print('graph module:', graph_module.src)
+            script = torch.jit.script(graph_module)
+
+            # make sure graph module and script module are both runanble
+            original_out = graph_module(input_value)
+            is_not_tuple_out = not isinstance(original_out, tuple)
+            script_out = script(input_value)
+            self.assertEqual(
+                (original_out - script_out).abs().max(), 0,
+                'Reslut of original graph module and script module does not match')
+
+            # set to train just before quantization
+            if mode != 'static':
+                model.train()
+
+            graph_module = fuse(graph_module)
+            quantizer = Quantizer()
+            prepared = quantizer.prepare(graph_module, qconfig_dict)
+
+            if mode == 'ddp':
+                mp.spawn(run_ddp,
+                         args=(world_size, prepared),
+                         nprocs=world_size,
+                         join=True)
```

Comment:
I feel like for DDP we can test that it runs without errors on a toy model, and deduce the correctness of real models from the guarantees of FX and DDP provide.  So, not sure if we need to test torchvision in DDP specifically as a part of the test suite, although I'm flexible

Reply:
this is e2e test and we have a reference implementation of eager mode so I think it is valuable.

Reply:
we can also have unit tests for ddp to just make it runs as well

Reply:
sure, would it be ok to separate DDP testing into its own file then, so that the reader can ignore it unless they are specifically working on it

I'm not sure if we need to test all the models with DDP either, testing one model which contains SyncBatchNorm + a backwards pass + the guarantees of DDP will be enough to verify all models are correct if one is correct

Reply:
I can make ddp just test one model in this test case, splitting it to a separate file would probably mean copying the other code and I think we should avoid that